src/pin_git_modules.py

```diff
@@ -0,0 +1,53 @@
+import subprocess as sp
```

Comment:
@ShiboXing do you know why this is added to src?

Also this code requires some comments/docstrings - I think for now we should add this to a build artifacts directory rather than the source directory